library/utils.c

```diff
@@ -126,15 +122,45 @@ static char elapsed_buffer[32];
 static uint64_t clock_offset;
 
 static uv_prepare_t log_flusher;
-static log_writer logger = default_log_writer;
+static log_writer logger = NULL;
+static void init_debug(uv_loop_t *loop);
 
-void ziti_set_log(log_writer log_func, uv_loop_t *loop) {
+void ziti_log_init(uv_loop_t *loop, int level, log_writer log_func) {
     init_debug(loop);
-    uv_mbed_set_debug(ziti_debug_level, uv_mbed_logger);
-    logger = log_func;
+
+    if (level == ZITI_LOG_DEFAULT_LEVEL)
+        level = ziti_log_level; // in case it was set before
+
+    ziti_log_set_level(level);
+
+    if (log_func == NULL) {
+        // keep the logger if it was already set
+        ziti_log_set_logger(logger ? logger : default_log_writer);
+    } else {
+        ziti_log_set_logger(log_func);
+    }
 }
 
-void init_debug(uv_loop_t *loop) {
+void ziti_log_set_level(int level) {
+    if (level == ZITI_LOG_DEFAULT_LEVEL) {
+        char *lvl = getenv("ZITI_LOG");
```

Comment:
i feel like this would be hard to track down if didn't know about it. maybe this should produce a log message of its own as to why the log level is set. or - why not just drop ZITI_LOG entirely?